libecole/tests/src/reward/test-nnodes.cpp

```diff
@@ -0,0 +1,85 @@
+#include <memory>
+#include <string>
+#include <tuple>
+
+#include <catch2/catch.hpp>
+
+#include "ecole/environment/branching.hpp"
+#include "ecole/observation/nothing.hpp"
+#include "ecole/reward/nnodes.hpp"
+
+#include "conftest.hpp"
+#include "reward/unit-tests.hpp"
+
+using namespace ecole;
+
+TEST_CASE("NNodes unit tests", "[unit][reward]") {
+	reward::unit_tests(reward::NNodes{});
+}
+
+TEST_CASE(
+	"NNodes returns the difference in the total number of processed nodes between two states",
+	"[reward]") {
+
+	auto reward_func = reward::NNodes{};
+	auto model = get_solving_model();
+	reward_func.reset(model);
+
+	SECTION("NNodes is positive") { REQUIRE(reward_func.obtain_reward(model) >= 0); }
```

Comment:
Actually it is 0 here, so the behavior is the same as the next test. See also discussion in #68.

How about, branch once, then check that the `reward == 1`

Reply:
Agreed, the test is trivial

Reply:
So now the reverse control is part of the `Model` so it is completely transparent to use.
The following start a reverse control, presolve... and stops on the root node:
```cpp
model.solve_iter();
```
Then you can continue branching like so. Each call to `solve_iter_branch` branch an solve until the next node 
```cpp
while (!model.solve_iter_is_done()) {
    model.solve_iter_branch(/*some SCIP_VAR* */)
}
```
Everything gets properly managed and deallocated automatically, so this is really all there is :)